lib/compress.js

```diff
@@ -148,10 +148,19 @@ merge(Compressor.prototype, {
             node.process_expression(true);
         }
         var passes = +this.options.passes || 1;
-        for (var pass = 0; pass < passes && pass < 3; ++pass) {
+        var last_count;
+        for (var pass = 0; pass < passes; pass++) {
             if (pass > 0 || this.option("reduce_vars"))
                 node.reset_opt_flags(this, true);
             node = node.transform(this);
+            if (passes > 1) {
+                var count = 0;
+                node.walk(new TreeWalker(function() {
+                    count++;
+                }));
+                if (last_count == count) break;
```

Comment:
Yes, but there are other transforms going on at the same time in a given pass that'll influence the node count. Statistically it will never come up except in contrived hand-made cases.

Here's the thing - I'd like to use `passes=100` and not worry about the tree never reaching steady state as was the case in the past with a number of different transforms.

My take is that in 99.9% of the cases equality and inequality comparisons will yield the same number of passes, but in rare non-steady state cases equality comparisons will result in the maximum number of passes making uglify very slow on large JS inputs. I'd rather err on the side of running uglify quicker rather than having uglify run several times slower just to shave off a few bytes.